android/src/main/java/com/ai/ChatState.kt

```diff
@@ -30,7 +31,44 @@ class Chat(modelConfig: ModelConfig, modelDir: File) {
     }
   }
 
-  interface ChatStateCallback {
+  fun streamResponse(messages: MutableList<ChatCompletionMessage>, callback: StreamCallback) {
+    executorService.submit {
+      viewModelScope.launch {
+        val chatResponse = engine.chat.completions.create(messages = messages, stream_options = OpenAIProtocol.StreamOptions(include_usage = true))
+
+        var finishReasonLength = false
+        var streamingText = ""
+
+        for (res in chatResponse) {
+
+              for (choice in res.choices) {
+                choice.delta.content?.let { content ->
+                  streamingText += content.asText()
+                }
+                choice.finish_reason?.let { finishReason ->
+                  if (finishReason == "length") {
+                    finishReasonLength = true
+                  }
+                }
+              }
+
+              callback.onUpdate(streamingText)
+              if (finishReasonLength) {
+                streamingText += " [output truncated due to context length limit...]"
+                callback.onUpdate(streamingText)
+              }
```

Comment:
Hm, in future probably other kind of communication we'd need to leverage other than adding this to user-facing stream.